src/backend/distributed/transaction/transaction_management.c

```diff
@@ -478,6 +491,74 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 }
 
 
+/*
+ * CloseAllInProgressConnections closes all connections of in progress transactions.
+ */
+static void
+CloseAllInProgressConnections(void)
+{
+	dlist_iter iter;
+	dlist_foreach(iter, &InProgressTransactions)
+	{
+		MultiConnection *connection = dlist_container(MultiConnection,
+													  transactionNode,
+													  iter.cur);
+
+		CloseConnection(connection);
```

Comment:
I think it'd be slightly nicer to set `connection->forceCloseAtTransactionEnd = true;` and let AfterXactConnectionHandling handle the closure.